project-koku/masu
# Adjusting OCP tier calculation method and combining request and usage charges.

**Overview**
Tier calculations are now done on the basis that each applicable tier rate is used for the appropriate portion of the usage.

In addition to the tier recalculation, the CPU and Memory charges are now a summation of their respective usage and request charges.

**Testing**
1.  Add OCP provider
2.  Add CPU Usage rate for provider
3. Add Memory Usage rate for provider
4.  Generate OCP data 
5. Ingest and process OCP data with masu
6. Verify CPU and Memory koku API response
7.  Add CPU and Memory request rates to provider.
8. Refresh charges with masu endpoint
9.  Re-run same Koku API queries from step 6 and verify that the charges have been updated.
10. Remove all rates and refresh masu data.  Verify that charge is 0 in Koku API

**Test Results**
[masu_tiers_recalc_ut.txt](https://github.com/project-koku/masu/files/2654864/masu_tiers_recalc_ut.txt)

Closes #281
